common4j/src/main/com/microsoft/identity/common/java/nativeauth/providers/responses/signin/SignInTokenApiResponse.kt

```diff
@@ -81,6 +82,16 @@ class SignInTokenApiResponse(
                             correlationId = correlationId
                         )
                     }
+                    errorCodes[0].isPasswordChangeRequired() -> {
+                        val customDescription = "Password change is required, which can't be fulfilled as part of this flow. " +
```

Comment:
Is this necessary? You're already adding this information in `ExceptionAdapter`. Also, I'd recommend keeping the original API error description and not altering them. Developer should have a way to access to the raw API response.

Reply:
I think the "ExceptionAdapter" class handle the error when we're refreshing the access token, while in this case we handle the scenario of obtaining a new token. Correct me if I'm wrong.
About the error description, I'm adding the description from API at the end of the custom message. Do you suggest adding the custom string in a different place?

Reply:
> Correct me if I'm wrong.

You're probably right, but I'm not 100% sure - you'll have to run tests.

Reply:
Ok, I did a test, and this code is executed during the signIn. So, my assumption was correct